pkg/proxy/winkernel/proxier.go

```diff
@@ -644,7 +648,14 @@ func NewDualStackProxier(
 
 	// Return a meta-proxier that dispatch calls between the two
 	// single-stack proxier instances
-	return metaproxier.NewMetaProxier(ipv4Proxier, ipv6Proxier), nil
+	return metaproxier.NewMetaProxier(
+		ipv4Proxier,
+		ipv6Proxier,
+		hostname,
+		[]string{"0.0.0.0", "::0"}, /* dual stack proxy will work on 0.0.0.0 and ::0 */
```

Comment:
0.0.0.0 and ::0 are the same for golang in terms of socket binding, why is this hardcoded by the way?

Reply:
That's true on Linux. I'm not sure it's true on Windows. (It requires OS support; `net.Listen` is constrained to only listen on a single socket, so if the OS can't provide a single socket that accepts both IPv4 and IPv6 connections, then `net.Listen` can't either.)

FWIW `::0` is not in canonical form. (Should be `::`.)